python/ngen_conf/src/ngen/config/hydrofabric.py

```diff
@@ -0,0 +1,74 @@
+
+from typing import List, Literal
+from pydantic import BaseModel, validator
+from geojson_pydantic import Feature, Point,  MultiPolygon
+
+def check_in_accepted(val, accepted):
+    if val not in accepted:
+        raise ValueError(f'Not an acceptable option. Options are {accepted}')
+    
+class NexusFeatureProperty(BaseModel):
+    toid: str
+
+    @validator('toid')
+    def check_prefix(cls,val):
+        assert val[:3] == 'cat', "toid must be a cat id"
+
+NexusFeature = Feature[Point, NexusFeatureProperty]
+
+class CatchmentFeatureProperty(BaseModel):
+    toid: str
+    areasqkm : float
+
+    @validator('toid')
+    def check_prefix(cls,val):
+        assert val[:3] == 'nex', "toid must be a nexus id"
+
+CatchmentFeature = Feature[MultiPolygon, CatchmentFeatureProperty]
+
+class CRSProperty(BaseModel):
```

Comment:
This is [optional in the GeoJSON spec](https://gist.github.com/sgillies/1233327#file-geojson-spec-1-0-L113), so I think it is a little overkill and potentially error prone to validate this. I feel even better about not validating it given that [rfc7946](https://www.rfc-editor.org/rfc/rfc7946), the GeoJSON Format notes in 3.4: 

> The coordinate reference system for all GeoJSON coordinates is a
   geographic coordinate reference system, using the World Geodetic
   System 1984 (WGS 84) [[WGS84](https://www.rfc-editor.org/rfc/rfc7946#ref-WGS84)] datum, with longitude and latitude units
   of decimal degrees.  This is equivalent to the coordinate reference
   system identified by the Open Geospatial Consortium (OGC) URN
   urn:ogc:def:crs:OGC::CRS84

 I appreciate the thoroughness though :smile:!

Reply:
Yup. Figured we may want to validate that field, but we can just leave it out if not.